### ISSUE TITLE
Error handling does not catch authorization error

### ISSUE THREAD

**Issue by c-herring** [Labelbox/labelbox-python#65]
https://github.com/Labelbox/labelbox-python/blob/7b3ae4a9ffaf3a2c45e47b77c5050c8be821b9eb/labelbox/client.py#L153

When errors is of the form:
`[{'message': 'You are unauthorized to perform this action', 'locations': [], 'path': ['setProjectMembership'], 'extensions': {'code': 'INTERNAL_SERVER_ERROR', 'exception': {'context': {'userId': 'userid', 'organizationId': 'orgid'}, 'code': 'AUTHORIZATION_ERROR'}}}]
obj:  {'message': 'You are unauthorized to perform this action', 'locations': [], 'path': ['setProjectMembership'], 'extensions': {'code': 'INTERNAL_SERVER_ERROR', 'exception': {'context': {'userId': 'userid', 'organizationId': 'orgid'}, 'code': 'AUTHORIZATION_ERROR'}}}`

(eg using setProjectMembership for a user who is admin and therefore cannot be removed from a project)

Error handling checks to see if value under `code` or `extensions` is equal to `AUTHORIZATION_ERROR`

But `code` does not even exist at the top level.

`extensions` is a dict so obviously that does not equal the string `AUTHORIZATION_ERROR`

This is true for all other error handling for the response of the form listed above.

So either this response is different to other responses (I haven't checked other responses) and error handling is just broken in this case.

Or, all responses are like this and error handling is just broken in general.

**Comment by rllin** [comment]
@c-herring what is the error you are encountering?  could you paste the stacktrace?

otherwise, if I understand you correctly, i believe the `check_errors` call you link to should traverse your given list of errors and then NOT match.

and it will match later for this call
https://github.com/Labelbox/labelbox-python/blob/7b3ae4a9ffaf3a2c45e47b77c5050c8be821b9eb/labelbox/client.py#L179

and return an `INTERNAL_SERVER_ERROR` error instead

is that what is actually happening?

**Comment by rllin** [comment]
also @c-herring are you on a free tier by any chance?

**Comment by c-herring** [comment]
I am not free teir, the email I use for my Labelbox account is Chris.Herring@invetech.com.au

No, it raises a generic LabelBox unknown error here:
https://github.com/Labelbox/labelbox-python/blob/7b3ae4a9ffaf3a2c45e47b77c5050c8be821b9eb/labelbox/client.py#L187
because it does not match anything through all those error checks. This is because the check_errors function only looks one level deep in the error dictionary. Whereas `INTERNAL_SERVER_ERROR` is two levels deep (error['extensions']['code']) and `AUTHORIZATION_ERROR` is (error['extensions']['exception']['context]'['code'])

Note: this below error is generated in release version 2.3 
The same problem happens in 2.6 however because it is wrapped in retry logic, it just gets stuck in a retry loop until it finally raises some generic error. So I am using 2.3 so that it actually raises a Labelbox error and I can catch the error in my own code as a workaround.

`WARNING:Unparsed errors on query execution: [{'message': 'You are unauthorized to perform this action', 'locations': [], 'path': ['setProjectMembership'], 'extensions': {'code': 'INTERNAL_SERVER_ERROR', 'exception': {'context': {'userId': 'cjyi0u6iy4xai0701o95bhmo4', 'organizationId': 'cjyi0u6ih59pu09447a6tolb0'}, 'code': 'AUTHORIZATION_ERROR'}}}]
C:\Anaconda3-5_2_0\envs\ml-tf220-pip\lib\site-packages\labelbox\client.py in execute(self, query, params, timeout)
    168             logger.warning("Unparsed errors on query execution: %r", errors)
    169             raise labelbox.exceptions.LabelboxError(
--> 170                 "Unknown error: %s" % str(errors))
    171 
    172         return response["data"]

LabelboxError:  `

**Comment by rllin** [comment]
thanks, this is useful.

how are you making `setProjectMembership` calls?  i believe those are not directly supported by the sdk?  my suspicion is the same as yours, that the errors being returned are just not standardized which is why you may be seeing this with `setProjectMembership` (tho i will have to double check this on the backend)

**Comment by c-herring** [comment]
Oh sorry, Actually it is not broken!

As it turns out, 2.6 does throw INTERNAL_SERVER_ERROR, however that just goes through the retry loop until it eventually raises some other error.
I went back to 2.3 so avoid that retry logic so that I could catch when I did not have permission to remove a user from a project and 2.3 actually does not have a check for INTERNAL_SERVER_ERROR!

Sorry about that, seems it works as expected. 

I am just calling setProjectMembership using client.execute as it is a convenient way to interact with graphql. I use the python sdk for other things where I can.

Seems the problem I have reached is that client.execute has gotten a bit more complexity and now does ot really support general graphql calls, presumable since the graphql call is not supported by the sdk it has not has its return standardized. 

Time to fork 2.3 I guess :)
